### ISSUE TITLE
Make a ZIM of http://cyclowiki.org/wiki/

### ISSUE THREAD

**Issue by kelson42** [openzim/zim-requests#9]
_From @kelson42 on August 26, 2018 14:15_

_Copied from original issue: openzim/mwoffliner#363_

**Comment by kelson42** [comment]
`mwoffliner --mwUrl="http://cyclowiki.org" --adminEmail="kelson@kiwix.com" --localParsoid --verbose`

dies with

```
Executing command : pngquant --verbose --strip --nofs --force --ext=".ag2w2.png" "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/video-icon.png" && advdef -q -z -4 -i 5 "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/video-icon.ag2w2.png" && if [ $(stat -c%s "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/video-icon.ag2w2.png") -lt $(stat -c%s "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/video-icon.png") ]; then mv "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/video-icon.ag2w2.png" "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/video-icon.png"; else rm "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/video-icon.ag2w2.png"; fi
Executing command : pngquant --verbose --strip --nofs --force --ext=".2db4l.png" "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/document-icon.png" && advdef -q -z -4 -i 5 "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/document-icon.2db4l.png" && if [ $(stat -c%s "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/document-icon.2db4l.png") -lt $(stat -c%s "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/document-icon.png") ]; then mv "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/document-icon.2db4l.png" "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/document-icon.png"; else rm "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/document-icon.2db4l.png"; fi
Executing command : pngquant --verbose --strip --nofs --force --ext=".b7rl0.png" "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/audio-icon.png" && advdef -q -z -4 -i 5 "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/audio-icon.b7rl0.png" && if [ $(stat -c%s "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/audio-icon.b7rl0.png") -lt $(stat -c%s "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/audio-icon.png") ]; then mv "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/audio-icon.b7rl0.png" "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/audio-icon.png"; else rm "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/audio-icon.b7rl0.png"; fi
Executing command : pngquant --verbose --strip --nofs --force --ext=".a5s3f.png" "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/user-icon.png" && advdef -q -z -4 -i 5 "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/user-icon.a5s3f.png" && if [ $(stat -c%s "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/user-icon.a5s3f.png") -lt $(stat -c%s "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/user-icon.png") ]; then mv "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/user-icon.a5s3f.png" "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/user-icon.png"; else rm "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/user-icon.a5s3f.png"; fi
Successfuly optimized /srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/video-icon.png
Successfuly optimized /srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/audio-icon.png
Successfuly optimized /srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/document-icon.png
Successfuly optimized /srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/watch-icons.png
Saving favicon.png...
Downloading http://cyclowiki.org/w/api.php?action=query&meta=siteinfo&format=json...
Executing command : gifsicle --verbose --colors 64 -O3 "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/watch-icon-loading.gif" -o "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/watch-icon-loading.tz4ir.gif" && if [ $(stat -c%s "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/watch-icon-loading.tz4ir.gif") -lt $(stat -c%s "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/watch-icon-loading.gif") ]; then mv "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/watch-icon-loading.tz4ir.gif" "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/watch-icon-loading.gif"; else rm "/srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/watch-icon-loading.tz4ir.gif"; fi
Successfuly optimized /srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/user-icon.png
Successfuly optimized /srv/kiwix-maintenance/mwoffliner/tmp/cyclowiki_ru_all_2018-09/s/watch-icon-loading.gif
TypeError [ERR_INVALID_ARG_TYPE]: The "url" argument must be of type string. Received type undefined
    at Url.parse (url.js:150:11)
    at Object.urlParse [as parse] (url.js:144:13)
    at /usr/local/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2083:29
    at async.retry (/usr/local/lib/node_modules/mwoffliner/lib/Downloader.js:176:7)
    at /usr/local/lib/node_modules/mwoffliner/node_modules/async/lib/async.js:676:51
    at /usr/local/lib/node_modules/mwoffliner/node_modules/async/lib/async.js:726:13
    at /usr/local/lib/node_modules/mwoffliner/node_modules/async/lib/async.js:52:16
    at /usr/local/lib/node_modules/mwoffliner/node_modules/async/lib/async.js:264:21
    at /usr/local/lib/node_modules/mwoffliner/node_modules/async/lib/async.js:44:16
    at /usr/local/lib/node_modules/mwoffliner/node_modules/async/lib/async.js:723:17

```

**Comment by kelson42** [comment]
_From @ISNIT0 on September 18, 2018 9:39_

No logoUrl is returned by the api for this site (http://cyclowiki.org/w/api.php?action=query&meta=siteinfo&format=json)

What should be done in this situation?
Use a placeholder?

**Comment by kelson42** [comment]
@ISNIT0 It should stop with a proper error msg inviting to use `--customZimFavicon`

**Comment by kelson42** [comment]
Impacted by https://github.com/openzim/mwoffliner/issues/387

**Comment by kelson42** [comment]
Done at https://farm.openzim.org/recipes/cyclowiki

**Comment by benoit74** [comment]
Still not published

**Comment by benoit74** [comment]
We are blocked by Cloudflare, no matter which User-Agent / Referer we pass, Cloudflare still blocks us on first call to `sanitize_mwUrl` (i.e. probably first call ever) on most workers.

**Comment by Popolechien** [comment]
So close as `wontfix` or do we wait for a decision on [mwoffliner#2134](https://github.com/openzim/mwoffliner/issues/2134)?

**Comment by benoit74** [comment]
#2134 might help, so probably worth to wait

**Comment by benoit74** [comment]
https://github.com/openzim/mwoffliner/issues/2134 is fixed, but we are now blocked by https://github.com/openzim/mwoffliner/issues/2091

**Comment by kelson42** [comment]
New attempt with 1.15 at https://farm.openzim.org/recipes/cyclowiki.org_ru_all

**Comment by benoit74** [comment]
This is clearly blocked by cloudflare. Closing issue as we have no way currently to work around this blockage. Should someone have an idea about how to workaround this, feel free to ask for reopen.

**Comment by benoit74** [comment]
Last run is moving fine: https://farm.openzim.org/pipeline/ab1c0168-ca92-440d-be10-7c83de35231d

I'm pretty sure https://github.com/openzim/mwoffliner/pull/2395 was the change required.

**Comment by kelson42** [comment]
Last attempt unfortunately died during the `maxi` computation. Reason is not that clera to me why... but I have relaunched with `dev` and slow-down the speed. See https://farm.openzim.org/pipeline/cd2d2a5c-5d29-4b44-9eec-281af7e0cdd6

**Comment by kelson42** [comment]
Last attempt was successful, see https://farm.openzim.org/recipes/cyclowiki.org_ru_all

@benoit74 @Popolechien ZIM files are available to final review at https://dev.library.kiwix.org/#lang=&q=cyclowiki. Are we good to go to production?

**Comment by Popolechien** [comment]
LGTM but I'm very very surprised by the size discrepancy between `mini`/`nopic` and `maxi` (905/914MB vs. 25GB). Should we also maybe do without the `mini` version?

Other than that, spotted this at [Древнегреческий язык](https://dev.library.kiwix.org/viewer#cyclowiki.org_ru_all_maxi_2025-07/%D0%94%D1%80%D0%B5%D0%B2%D0%BD%D0%B5%D0%B3%D1%80%D0%B5%D1%87%D0%B5%D1%81%D0%BA%D0%B8%D0%B9_%D1%8F%D0%B7%D1%8B%D0%BA)

<img width="999" height="693" alt="Image" src="https://github.com/user-attachments/assets/e50eaa89-b18d-4994-9ede-bfc07459e69b" />



Source is [here](https://cyclowiki.org/wiki/%D0%94%D1%80%D0%B5%D0%B2%D0%BD%D0%B5%D0%B3%D1%80%D0%B5%D1%87%D0%B5%D1%81%D0%BA%D0%B8%D0%B9_%D1%8F%D0%B7%D1%8B%D0%BA) and shows those are two embedded youtube videos. 

Not a blocker IMHO, but worth flagging. Should I open a separate issue?

**Comment by benoit74** [comment]
> Should I open a separate issue?

Yes, please do in openzim/mwoffliner, worth to fix (and especially decide how to fix this)

**Comment by benoit74** [comment]
>  I'm very very surprised by the size discrepancy between mini/nopic and maxi (905/914MB vs. 25GB). Should we also maybe do without the mini version?

I'm sorry, I don't get it, what is the problem of having a very small `mini` ZIM if images on the wiki are very big / numerous and lead to a very big `maxi`?

**Comment by Popolechien** [comment]
Yeah that what poorly phrased as these are two separate questions:

- Seeing how close in size `mini` and `nopic`are, I do not see the value of generating a `mini` (the distinct flavours/size are confusing to users, which is compounded by the fact that filtering is not really optimal)
- I am suprised at the size discrepancy (on WP for instance `nopic` would be half the size of `maxi` not 4% like here so I was wondering if this was a bug or feature (ie, is the wiki so media-rich that removing them makes a world of a difference).

**Comment by benoit74** [comment]
We are already only generating `nopic` and `maxi`, `mini` is already not generated AFAIK (or I miss something).

Regarding size discrepancy, I do not have sufficient tooling/time to investigate details ATM.
